eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -77,7 +77,7 @@ jobs:
         OSVmImage: MMSUbuntu18.04
         vcpkg.deps: 'curl[ssl] paho-mqtt cmocka'
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
-        build.args: '-DTRANSPORT_CURL=ON -DTRANSPORT_PAHO=ON -DAZ_PLATFORM_IMPL=POSIX -DPRECONDITIONS=OFF -DUNIT_TESTING=ON -DUNIT_TESTING_MOCKS=ON'
+        build.args: '-DTRANSPORT_CURL=ON -DTRANSPORT_PAHO=ON -DAZ_PLATFORM_IMPL=POSIX -DPRECONDITIONS=OFF -DUNIT_TESTING=ON -DADDRESS_SANITIZER=ON -DUNIT_TESTING_MOCKS=ON'
```

Comment:
Do we need to turn it on both here and on line 70, or is just one of Release or Debug build type enough? Other than that there's not much difference between the two legs, and looks like applying Asan on either one gets us the Linux coverage.

Reply:
No, that's your choice. The idea is to catch any memory errors that only occur in one or the other. Either errors in code that was removed by the optimizer in debug mode or errors that never get hit unless the compiler does some optimization (if you write a check that happens to be UB or something).